app/src/main/java/protect/card_locker/ManageGroupActivity.java

```diff
@@ -0,0 +1,264 @@
+            }
+            mCurrentGroupName = inputSavedInstanceState.getString("mCurrentGroupName");
+        }
+
+        ActionBar actionBar = getSupportActionBar();
+        if (actionBar == null){
+            throw(new RuntimeException("mActionBar is not expected to be null here"));
+        }
+        actionBar.setDisplayHomeAsUpEnabled(true);
+        actionBar.setDisplayShowHomeEnabled(true);
+
+        saveButton.setOnClickListener(v -> {
+            String currentGroupName = mGroupNameText.getText().toString();
+            if(!currentGroupName.trim().equals(mGroup._id)){
+                if(!mGroupNameNotInUse) {
+                    Toast toast = Toast.makeText(getApplicationContext(), R.string.group_name_already_in_use, Toast.LENGTH_SHORT);
+                    toast.show();
+                    return;
+                }
+                if(currentGroupName.trim().length() == 0){
+                    Toast toast = Toast.makeText(getApplicationContext(), R.string.group_name_is_empty, Toast.LENGTH_SHORT);
+                    toast.show();
+                    return;
+                }
+            }
+
+            mAdapter.commitToDatabase();
+            Toast toast = Toast.makeText(getApplicationContext(), R.string.group_updated, Toast.LENGTH_SHORT);
+            if(!currentGroupName.trim().equals(mGroup._id)){
+                mDB.updateGroup(mGroup._id, currentGroupName.trim());
+            }
+            toast.show();
+            finish();
+        });
+        mHelpText.setText(getResources().getText(R.string.noGiftCardsGroup));
+    }
+
+    private void checkIfGroupNameIsInUse(){
+        mGroupNameNotInUse = false;
+        String currentGroupName = mGroupNameText.getText().toString();
+        if (!mGroup._id.equals(currentGroupName.trim())) {
+            Group group = mDB.getGroup(currentGroupName.trim());
+            if (group != null) {
+                mGroupNameNotInUse = false;
+                mGroupNameText.setError(getResources().getText(R.string.group_name_already_in_use));
+            } else {
+                mGroupNameNotInUse = true;
+            }
+        }
+    }
+
+    @Override
+    protected void onResume()
+    {
+        super.onResume();
+
+        setTitle(getString(R.string.editGroup, mGroup._id));
+
+        if (mCurrentGroupName == null){
+            mGroupNameText.setText(mGroup._id);
+        }else{
+            mGroupNameText.setText(mCurrentGroupName);
+            checkIfGroupNameIsInUse();
+        }
+
+        mGroupNameText.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+            }
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+            }
+
+            @Override
+            public void afterTextChanged(Editable s) {
+                mGroupNameText.setError(null);
```

Comment:
Maybe just move this into checkIfGroupNameIsInUse? So you don't empty the error and then set it again?